### PR TITLE
proxyd updates from upstream

### PR DESCRIPTION
**Pull request by InoMurko**
A number of updates have been pulled from upstream Optimism repo.

## Overview

Describe what your Pull Request is about in a few sentences.
1. testing is now done in cirleci (there were no tests before)
2. the image is now build with develop and with release

Describe your changes and implementation choices. More details make PRs easier to review.

3. support for pattern matchin in exempt origins. This should help with some of the issues Synthetix and others are seeing.
4. https://github.com/ethereum-optimism/optimism/pull/3700
5. https://github.com/ethereum-optimism/optimism/pull/3704
6. https://github.com/ethereum-optimism/optimism/pull/3681
7. https://github.com/ethereum-optimism/optimism/pull/3679
8. https://github.com/ethereum-optimism/optimism/pull/3683
9. https://github.com/ethereum-optimism/optimism/pull/3545
10. https://github.com/ethereum-optimism/optimism/pull/3544
11. https://github.com/ethereum-optimism/optimism/pull/3503
12. https://github.com/ethereum-optimism/optimism/pull/3482
13. https://github.com/ethereum-optimism/optimism/pull/3471
14. https://github.com/ethereum-optimism/optimism/pull/3165
15. https://github.com/ethereum-optimism/optimism/pull/3166
16. https://github.com/ethereum-optimism/optimism/pull/3110
17. https://github.com/ethereum-optimism/optimism/pull/2801
18. https://github.com/ethereum-optimism/optimism/pull/2711
19. https://github.com/ethereum-optimism/optimism/pull/2628

## Testing

Unit testing and integration testing in circleci

### CODE REVIEW

**Review by wsdt — APPROVED** [review]
Looks good to me

File: go/proxyd/proxyd.go [hunk 1]
```diff
@@ -147,17 +292,82 @@ func Start(config *Config) error {
...
+		}
+		log.Info("goodbye")
+	}, nil
 }
 
 func secondsToDuration(seconds int) time.Duration {
 	return time.Duration(seconds) * time.Second
 }
+
+func configureBackendTLS(cfg *BackendConfig) (*tls.Config, error) {
```
Comment (rupertbg) [inline go/proxyd/proxyd.go:318]: It looks like this perhaps adds support for TLS to the backends?

Comment (InoMurko) [reply go/proxyd/proxyd.go:318]: Yep!

**Comment by InoMurko** [comment]
@Boyuan-Chen @mmontour1306 @souradeep-das

**Review by souradeep-das — APPROVED** [review]
LGTM 🚀

**Review by boyuan-chen — APPROVED** [review]

**Pull request merged**